pkg/directory/types/farm.go

```diff
@@ -252,13 +266,51 @@ func FarmUpdate(ctx context.Context, db *mongo.Database, id schema.ID, farm Farm
 }
 
 // FarmIPReserve reserves an IP if it's only free
-func FarmIPReserve(ctx context.Context, db *mongo.Database, farm schema.ID, ip schema.IPCidr, reservation schema.ID) error {
+func FarmIPReserve(ctx context.Context, db *mongo.Database, farm schema.ID, publicIpWorkload *workloads.PublicIP) error {
 	col := db.Collection(FarmCollection)
-	// filter using 0 reservation id (not reserved)
-	filter := FarmFilter{}.WithID(farm).WithIP(ip, 0)
+
+	// default reservation id to look for
+	// 0 = free
+	reservationID := 0
+
+	// filter on farms where a farm has this ip
+	filter := FarmFilter{}.WithID(farm).WithIP(publicIpWorkload.IPaddress)
+	farmWithIP, err := filter.Get(ctx, db)
+	if err == nil {
```

Comment:
That's a weird check, why not `if err != nil {return err}` instead of having this huge if statement!